Separate "waiting period threshold" into own setting.

davidtwco opened the pull request:
<!-- What's this PR for?  (Just a link to an issue is fine.) -->
This PR follows-up on #12276 to separate the "waiting period threshold" based on [discussion in Zulip](https://chat.zulip.org/#narrow/stream/137-feedback/topic/too.20new.20to.20add.20subscriber.20to.20stream/near/740683).

Quoting from the first commit:
 > This commit separates the `waiting_period_threshold` setting from
    the `create_stream_policy` setting, adding a new setting that the user
    can use to select a waiting period threshold.
>
> Both the invite to stream policy and create stream policy now have three
    options: admins only, members and admins, or members after waiting
    period/admins.

cc @rishig @timabbott 

**Testing Plan:** <!-- How have you tested? -->
I've updated the frontend tests (both node and casper) and tested manually.

**GIFs or Screenshots:** <!-- If a UI change.  See:
  https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
  -->
![image](https://user-images.githubusercontent.com/1295100/57480922-e3906f80-7298-11e9-90a0-3b1dc69884e7.png)

<!-- Also be sure to make clear, coherent commits:
  https://zulip.readthedocs.io/en/latest/contributing/version-control.html
  -->

timabbott commented:
I assume for "new users" we want to have it just be "All members with ..." -- admins shouldn't ever have that tag, I think?

rishig commented on static/js/admin.js:
```diff
@@ -19,6 +19,8 @@ var admin_settings_label = {
     realm_email_changes_disabled: i18n.t("Prevent users from changing their email address"),
     realm_avatar_changes_disabled: i18n.t("Prevent users from changing their avatar"),
+    realm_waiting_period_applies_to_create_stream: i18n.t("Creating streams"),
+    realm_waiting_period_applies_to_invite_to_stream: i18n.t("Inviting other members to streams"),
```
Let's make this "Adding users to streams", since we don't really have an "Inivtation" concept yet, and also this applies more broadly than just members.

davidtwco commented:
I've updated this PR after [discussion in Zulip](https://chat.zulip.org/#narrow/stream/137-feedback/topic/too.20new.20to.20add.20subscriber.20to.20stream/near/740683) with @rishig.

davidtwco commented:
I've added a new commit that changes the terminology used based [on this conversation](https://chat.zulip.org/#narrow/stream/137-feedback/topic/too.20new.20to.20add.20subscriber.20to.20stream/near/740791).

timabbott commented:
@rishig can you review the UX on this?

rishig commented:
Thanks David for your work on this! The commits had a number of changes mixed together, so I separated them out a bit and made a few wording and variable name changes as well.

The three commits ending in https://github.com/zulip/zulip/commit/c3af126766af8727bb689f8762791a2da3754f2a
have the changes to the code. Would you be able to take a look and also update the tests?

Once this is merged I can deal with updating and merging the documentation commits. 
Thanks again for your work on this!

davidtwco commented:
@rishig I've updated this PR with the commits from your branch (GitHub orders them strangely, but they look fine locally so I assume it is just a rendering issue).

I've fixed the tests so they pass after your changes.

rishig commented:
Tim, just making sure this isn't blocking on me?

timabbott commented:
Oh, I definitely thought this waiting on you, sorry about that!  Looking now.

timabbott commented on templates/zerver/help/configure-who-can-invite-to-streams.md:
```diff
@@ -3,17 +3,15 @@
-* **Organization administrators, and members with accounts older than the new user waiting period**
+* **Organization administrators**
+* **Organization administrators and all members**
+* **Organization administrators and full members**
```
"full members" here should be a link.

timabbott commented on templates/zerver/help/configure-who-can-create-streams.md:
```diff
@@ -2,27 +2,24 @@
-  for any `N`.
+* **Organization administrators**
+* **Organization administrators and all members**
+* **Organization administrators and full members**
```
Same here.

timabbott commented on static/styles/settings.scss:
```diff
@@ -1545,6 +1545,7 @@ body:not(.night-mode) #account-settings .custom_user_field .datepicker {
+#id_realm_waiting_period_setting,
 #id_realm_create_stream_policy,
 #id_realm_invite_to_stream_policy,
 #id_realm_org_join_restrictions,
```
We should probably restructure these to use a common CSS class; but probably not a blocker for merging :)

@pragatiagrawal31 @shubhamdhama FYI.
